File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/event/PlayerCommandPreprocessEvent.java
package top.likoslupus.cellulosesz.api.event;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Objects;

public final class PlayerCommandPreprocessEvent extends AbstractCancellableEvent {

    private final CellPlayer player;
    private String command;

    public PlayerCommandPreprocessEvent(
            CellPlayer player,
            String command
    ) {
        this.player = Objects.requireNonNull(player, "player");
        this.command = Objects.requireNonNull(command, "command");
    }

    public CellPlayer player() {
        return player;
    }

    public String command() {
        return command;
    }

    public void command(String command) {
        this.command = Objects.requireNonNull(command, "command");
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/event/PlayerCommandPreprocessEvent.kt
package top.likoslupus.cellulosesz.api.event

import top.likoslupus.cellulosesz.api.platform.CellPlayer

public class PlayerCommandPreprocessEvent(
    private val player: CellPlayer,
    private var command: String
) : AbstractCancellableEvent() {

    public fun player(): CellPlayer = player

    public fun command(): String = command

    public fun command(command: String) {
        this.command = command
    }

}
